components/net_traits/blob_url_store.rs

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use ipc_channel::ipc::IpcSender;
+use std::str::FromStr;
+use url::Url;
+use uuid::Uuid;
+
+/// Errors returns to BlobURLStoreMsg::Request
+#[derive(Clone, Serialize, Deserialize)]
+pub enum BlobURLStoreError {
+    /// Invalid UUID key
+    InvalidKey,
+    /// Invalid URL origin
+    InvalidOrigin,
+}
+
+/// Blob URL store entry, a packaged form of Blob DOM object
+#[derive(Clone, Serialize, Deserialize)]
+pub struct BlobURLStoreEntry {
+    /// MIME type string
+    pub type_string: String,
+    /// Some filename if the backend of Blob is a file
+    pub filename: Option<String>,
+    /// Size of content in bytes
+    pub size: u64,
+    /// Content of blob
+    pub bytes: Vec<u8>,
```

Comment:
This will eventually become an enum between bytes, on-disk file name, and slice, right?